android/runtime/common/src/java/org/appcelerator/kroll/KrollObject.java

```diff
@@ -78,8 +79,8 @@ public void setHasListenersForEventType(String event, boolean hasListeners)
 	 */
 	public void onEventFired(String event, Object data)
 	{
-		if (proxySupport != null) {
-			proxySupport.onEventFired(event, data);
+		if (proxySupport != null && proxySupport.get() != null) {
+			proxySupport.get().onEventFired(event, data);
```

Comment:
There's a race condition here where the weak reference can be collected between the 2 `get()` calls. It's slim, but it can happen. The following would solve it.

```java
KrollProxySupport proxy = (proxySupport != null) ? proxySupport.get() : null;
if (proxy != null) {
	proxy.onEventFired(event, data);
}
```

Not trying to be picky here, but I've seen Java code get burned by issues like this in the past. We should do the same in the `setHasListenersForEventType()` method.